packages/web-components/src/switch/switch.styles.ts

```diff
@@ -0,0 +1,176 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+  colorCompoundBrandBackground,
+  colorCompoundBrandBackgroundHover,
+  colorCompoundBrandBackgroundPressed,
+  colorNeutralBackgroundDisabled,
+  colorNeutralForeground1,
+  colorNeutralForeground3,
+  colorNeutralForeground3Hover,
+  colorNeutralForeground3Pressed,
+  colorNeutralForegroundDisabled,
+  colorNeutralForegroundInverted,
+  colorNeutralForegroundInvertedHover,
+  colorNeutralForegroundInvertedPressed,
+  colorNeutralStrokeAccessible,
+  colorNeutralStrokeAccessibleHover,
+  colorNeutralStrokeAccessiblePressed,
+  colorNeutralStrokeDisabled,
+  colorPaletteRedForeground3,
+  colorTransparentBackground,
+  curveEasyEase,
+  durationNormal,
+  fontFamilyBase,
+  fontSizeBase300,
+  fontWeightRegular,
+  lineHeightBase300,
+  spacingHorizontalS,
+  spacingHorizontalXS,
+  spacingVerticalS,
+  spacingVerticalXS,
+} from '../theme/design-tokens.js';
+
+export const styles = css`
+  ${display('flex')}
+
+  :host {
+    display: inline-flex;
+    align-items: center;
+    flex-direction: row-reverse;
+    outline: none;
+    user-select: none;
+  }
+  :host([label-position='before']) {
+    flex-direction: row;
+  }
+  :host([label-position='above']) {
+    flex-direction: column;
+    align-items: flex-start;
+  }
+  :host([disabled]) .label,
+  :host([readonly]) .label,
+  :host([readonly]) .switch,
+  :host([disabled]) .switch {
+    cursor: not-allowed;
+  }
+  :host .label {
+    position: relative;
+    color: ${colorNeutralForeground1};
+    line-height: ${lineHeightBase300};
+    font-size: ${fontSizeBase300};
+    font-weight: ${fontWeightRegular};
+    font-family: ${fontFamilyBase};
+    padding: ${spacingVerticalXS} ${spacingHorizontalXS};
+    cursor: pointer;
+  }
+  :host([required]) .label:after {
+    content: '*';
+    color: ${colorPaletteRedForeground3};
+    position: absolute;
+    right: -4px;
+  }
+  :host([disabled]) .status-message {
+    color: ${colorNeutralForegroundDisabled};
+  }
+  :host .switch {
+    position: relative;
+    box-sizing: border-box;
+    width: 40px;
+    height: 20px;
+    background-color: ${colorTransparentBackground};
+    border: 1px solid;
+    border-color: ${colorNeutralStrokeAccessible};
+    border-radius: 10px;
+    outline: none;
+    cursor: pointer;
+    margin: ${spacingVerticalS} ${spacingHorizontalS};
+  }
+  :host .switch * {
+    transition-duration: ${durationNormal};
+    transition-timing-function: ${curveEasyEase};
+    transition-property: transform;
+  }
+  :host(:not([disabled])) .switch:hover {
+    background: none;
+    border-color: ${colorNeutralStrokeAccessibleHover};
+  }
+  :host(:not([disabled])) .switch:active {
```

Comment:
Are you sure you need the `not` selectors? We should try and avoid them iff we can.

Reply:
If the hover and active states are still firing when `disabled` it would be better to just target the disabled overrides in multiple selector than using `not`.

```
:host([disabled]) .switch,
:host([disabled]) .switch:hover,
:host([disabled]) .switch:active {}
```